bundles/core/org.eclipse.smarthome.core.voice.test/src/test/java/org/eclipse/smarthome/core/voice/internal/SinkStub.java

```diff
@@ -28,19 +29,24 @@
  * some of the methods are not needed. That's why their implementation is left empty.
  *
  * @author Mihaela Memova - initial contribution
- *
  * @author Velin Yordanov - migrated from groovy to java
- *
  */
 public class SinkStub implements AudioSink {
 
     private boolean isStreamProcessed;
-    private boolean isUnsupportedAudioFormatExceptionExpected;
-    private Set<AudioFormat> supportedFormats = new HashSet<AudioFormat>();
+    private static final Set<AudioFormat> SUPPORTED_AUDIO_FORMATS = new HashSet<>();
```

Comment:
Perhaps we could drop the static block below and the unmodfiable set creation on the get method at all by using something like that here:

```Java
Collections.unmodifiableSet(Stream.of(AudioFormat.WAV, AudioFormat.MP3, ...).collect(Collectors.toSet())));
```

```Java
Collections.unmodifiableSet(new HashSet(Arrays.asList(AudioFormat.WAV, AudioFormat.MP3, ...)));
```

Reply:
Why do you think we should change that? Is the static initialization block not compliant to the guidelines?

Reply:
Everytime `getSupportedForrmats()` and `getSupportedStreams()` a new unmodifiable collection / view is created. Is there any need for? Why not create that unmodifiable view once and reuse that reference at every get... call?

Reply:
Makes sense. I misunderstood your first comment.